tools/packaging/common/istio-iptables.sh

```diff
@@ -83,41 +87,39 @@ function isIPv4() {
 #
 # Function return true if agrument is a valid ipv6 address
 #
-function isIPv6() {
+isIPv6() {
   local ipv6section="^[0-9a-fA-F]{1,4}$"
-  addr="$1"
+  addr=$(echo "$1" | sed -e "s/:/:${IFS}/g")
   number_of_parts=0
   number_of_skip=0
-  IFS=':' read -r -a addr <<< "$1"
-  if [ ${#addr[@]} -eq 0 ]; then
+  if [ "$(echo "${addr}" | awk -F"${IFS}"  '{print NF}')" -eq 0 ]; then
      return 1
   fi
-  for part in "${addr[@]}"; do
+  for part in ${addr}; do
+    part=${part%:}
     # check to not exceed number of parts in ipv6 address
-    if [[ ${number_of_parts} -ge 8 ]]; then
+    if [ ${number_of_parts} -ge 8 ]; then
         return 1
     fi
-    if [[ ${number_of_parts} -eq 0 ]] && ! [[ ${part} =~ ${ipv6section} ]]; then
+    if [ ${number_of_parts} -eq 0 ] && ! echo "${part}" | grep -q -E "${ipv6section}"; then
         return 1
     fi
-    if ! [[ ${part} =~ ${ipv6section} ]]; then
-       if ! [[ "$part" == "" ]]; then
+    if ! echo "${part}" | grep -q -E "${ipv6section}"; then
```

Comment:
wow, this set of iptables changes will need some serious dedicated review time.  How about a middle ground where istio-iptables.sh stays as is, and a distroless ash version is created in the interim so the main stable images are not potentially destablized?

Reply:
This would mean that one of variants (distroless,default) would have a significant lower test coverage than the other.
If you would like to go with the distroless variant for 1.2, you need to increase the test coverage for the `distroless/ash` variant as soon as possible. 

In my point of view, the modified docker images introduce the same amount of risk. Therefore I would try to roll out this switch to `distroless` as soon as possible.

Reply:
We added a test under `tests/scripts/istio-iptables-test.sh` which verifies, that the new variant is doing the same as the old one (except the two bugs #13979 and #13980)